jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/ValueReaderLocator.java

```diff
@@ -295,10 +304,27 @@ protected ValueReader arrayReader(Class<?> contextType, Class<?> arrayType) {
     }
 
     protected ValueReader enumReader(Class<?> enumType) {
+        // Call pojoDefinitionForDeserialization so that the annotation support extension can get custom names for
+        // enum values
+        POJODefinition def = null;
+        if (_readerModifier != null) {
+            def = _readerModifier.pojoDefinitionForDeserialization(_readContext, enumType);
+        }
+        Map<String, Object> byName =
+                JSON.Feature.ACCEPT_CASE_INSENSITIVE_ENUMS.isEnabled(_features) ? new TreeMap<String, Object>(
+                        String.CASE_INSENSITIVE_ORDER) : new HashMap<String, Object>();
         Object[] enums = enumType.getEnumConstants();
-        Map<String,Object> byName = new HashMap<String,Object>();
-        for (Object e : enums) {
-            byName.put(e.toString(), e);
+        if (def == null) {
+            for (Object e : enums) {
+                byName.put(e.toString(), e);
+            }
+        } else {
+            for (POJODefinition.Prop e : def.getProperties()) {
+                try {
+                    byName.put(e.name, e.field.get(null));
+                } catch (IllegalAccessException ignored) {
+                }
```

Comment:
Is this safe to ignore? Or would it cause hard-to-debug issues? (if access fails, code really can't succeed)

Reply:
I don't believe that enum values are able to be privated, so I'm fairly sure that this exception is impossible. If I did want to raise it up, how would that be done from this part of the code base? Just using an unchecked exception?